protocol/contracts/templegold/TempleGoldVesting.sol

```diff
@@ -0,0 +1,381 @@
+pragma solidity ^0.8.20;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Temple (templegold/TempleGoldVesting.sol)
+
+
+import { TempleElevatedAccess } from "contracts/v2/access/TempleElevatedAccess.sol";
+import { CommonEventsAndErrors } from "contracts/common/CommonEventsAndErrors.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { TempleMath } from "contracts/common/TempleMath.sol";
+import { ITempleGoldVesting } from "contracts/interfaces/templegold/ITempleGoldVesting.sol";
+
+
+/**
+ * @title TGLD Vesting
+ * @notice Vesting contract for contributors. Token is TGLD and allocations are set with createSchedules.
+ *  An account can have multiple vesting schedules. Vesting schedules can be revoked and canceled.
+ */
+contract TempleGoldVesting is ITempleGoldVesting, TempleElevatedAccess {
+    using SafeERC20 for IERC20;
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    /// @notice The owner of the TGLD funds
+    address public override fundsOwner;
+    /// @notice TGLD address
+    IERC20 public immutable override templeGold;
+    /// @notice Total TGLD vested and unclaimed
+    uint256 public override totalVestedAndUnclaimed;
+    /// @notice Schdules for recipients
+    mapping(bytes32 id => VestingSchedule schedule) public schedules;
```

Comment:
Out of interest, do you see a particular reason for using the id notation rather than a nested mapping?

eg vs:
```
mapping(address recipient => mapping(uint256 index => VestingSchedule schedule)) public schedules;
```

Just makes it a slightly less obvious API for consumers so wondering what the pros/cons are

Reply:
I used the id notation for the gas efficiency in read/write from storage.